cdap-common/src/main/java/co/cask/cdap/common/discovery/ResolvingDiscoverable.java

```diff
@@ -51,6 +51,10 @@ public InetSocketAddress getSocketAddress() {
 
   private InetSocketAddress resolve(InetSocketAddress bindAddress) {
     try {
+      // If domain of bindAddress is not resolvable, address of bindAddress is null.
+      if (bindAddress.getAddress() == null) {
```

Comment:
Seems like this is an unrelated change. Make sure you start with "develop" branch.